### PR TITLE
CIGAR parsing code for "variant" calls

### PR DESCRIPTION
**Pull request by BidaGu**
I wrote the variant calling code.

To make it working, I have to modify the _find_unique_isoform.py_ a bit to also output the reference isoform sequences. Readme files were updated accordingly.

A problem we may need to think about is handling of truncated cigar string. Because of the way pairwise alignment works, long mutation can be truncated into discontinuous mismatches & InDels. I implemented a refinement module that combines some of these mutations. But the issue is not fully resolved.

I also did some minor formatting of some of the old coding.

### CODE REVIEW

File: scripts/call_variants.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+## Output
+ref_id: reference sequence ID
+query_id: query sequence ID
+type: type of variant (snp/ins/del/delins)
+ref: variant sequence in reference
+alt: altered sequence
+ref_start: starting position of the variant in reference
+```
+#ref_id	query_id	type	ref	alt	ref_pos
```
Comment (evanbiederstedt) [inline scripts/call_variants.md:30]: I would put these in the standard VCF format:

```
CHR   REF_POS   REF ALT   TYPE   REF_ID.   QUERY_ID  
```

Please note in the comments that the position should be 1-based, which is standard for VCF (and please doublecheck that this is the case)

@BidaGu

Comment (evanbiederstedt) [reply scripts/call_variants.md:30]: Spoke with Bida about this just now. 

We'll update a markdown file to record the file format here.

Comment (evanbiederstedt) [reply scripts/call_variants.md:30]: Roughly, here's the summary:

We do a pairwise alignment between the REFERENCE and the QUERY.

* chromosome on reference/assembly
* starting position of REFERENCE on reference/assembly
* pos of variant on pairwise alignment
* Pair of IDs used for REF and QUERY, i.e. some sort of IDs for the pairwise alignment
* REF "allele" (whatever the bases are here; this is standard for VCF)
* ALT "allele" (whatever the bases are here; this is standard for VCF)
* the REFERENCE sequence
* the QUERY sequence

We should try to copy the VCF roughly, as this is what users will expect:

```
CHROM   POS   ID   REF   ALT 
```

https://en.wikipedia.org/wiki/Variant_Call_Format

**Review by evanbiederstedt — APPROVED** [review]
This looks reasonable to me, but I didn't try it

I'll approve

@BidaGu To make sure there's no issues, could you be able to add a basic test here for us to work with?

We'll need some (small) example data and the expected value after the code is run.

@cmatKhan and others will be able to help making this a decent unit tested piece of code

File: scripts/call_variants.md (hunk 1 above)
Comment (evanbiederstedt) [reply scripts/call_variants.md:30]: Here's a PR documenting this file format: https://github.com/collaborativebioinformatics/isocomp/pull/22

**Comment by cmatKhan** [comment]
Could we merge this into `develop` rather than `main`?

**Comment by cmatKhan** [comment]
could we change the target of this merge from `main` to `develop`?

**Comment by evanbiederstedt** [comment]
> Could we merge this into develop rather than main?

Thanks, I didn't notice this. 

Yes, @BidaGu could you do this?

If need be, you might have to port all of these changes to a branch off of develop. Otherwise, we'll have a mess. 

Please check the example of how I did this here: https://github.com/collaborativebioinformatics/isocomp/pull/22

We can discuss on slack as well in the open channel

**Pull request closed without merging**